src/main/js/helpers/axis.js

```diff
@@ -189,22 +190,38 @@ const getAxesScale = (axis, scale, config) => {
         scale.x,
         getAxisInfoOrientation(config.axis.x.orientation)
     );
+
+    if (
+        (config.ticksCount === undefined ||
+            config.ticksCount > constants.TICKSCOUNT_MAXLIMIT) &&
```

Comment:
why are we calculating ticks if they exceed max limit, can't we just go with max limit? please correct me if i am wrong.

Reply:
If you're talking about line 196, I'm checking to see if the consumer entered a value greater than `MAX_LIMIT`, e.g. 100 . If true, then the default number of ticks will be used.

Reply:
in that case looks like they are expecting for large tick count right, why can't we go with max limit. why are we choosing to go with default value, any particular reason?

Reply:
There's no technical difference between making tickCount the default or max value if given a value greater than max. This was just done so that `tickCount = undefined` and `tickCount > MAX_LIMIT` behavior will be consistent for simplicity.